client/lib/analytics/utils/cart-to-ga-purchase.ts

```diff
@@ -45,12 +46,22 @@ export function cartToGaPurchase(
 ): GaPurchase {
 	const cartInfoType = getCartInfoType( cartInfo );
 
+	// When using gtag.js, we can't access the `items` array to make custom events in GA4.
+	// To get around this limitation, we need to set this property on the top level purcahse object.
+	const containsYearlyOrHigherWPcomPlan = cartInfo.wpcomProducts?.some( ( product ) => {
+		if ( ! product.months_per_bill_period ) {
+			return false;
+		}
+		return product.months_per_bill_period >= 12;
+	} );
+
 	const { value, items } = parseCartInfo( cartInfo )[ cartInfoType ];
 	return {
 		transaction_id: orderId,
 		coupon: cart.coupon,
 		currency: 'USD', // we track all prices in USD
 		value,
 		items: items.map( ( product ) => productToGaItem( product, cart.currency ) ),
+		contains_yearly_or_higher_wpcom_plan: containsYearlyOrHigherWPcomPlan,
```

Comment:
I would suggest excluding `contains_yearly_or_higher_wpcom_plan` for carts other than wpcom:
```suggestion
		...( 'wpcom' === cartInfoType ? { contains_yearly_or_higher_wpcom_plan: containsYearlyOrHigherWPcomPlan } : {} ),
```

Reply:
Thanks! I've opted for just combining yearly and higher (incl 2y and 3y plans) for now, as the "and" filtering in Google isn't great, so this will be cleaner in the Google analytics setup. I've added the changes you suggested though for non-wpcom carts, which makes sense, as the prop is already optional in the type definition. :)